diff --git a/src/test/java/org/netbeans/modules/diff/builtin/PatchTest.java b/src/test/java/org/netbeans/modules/diff/builtin/PatchTest.java
index 027022e..3ca444c 100644
--- a/src/test/java/org/netbeans/modules/diff/builtin/PatchTest.java
+++ b/src/test/java/org/netbeans/modules/diff/builtin/PatchTest.java
@@ -3,9 +3,7 @@ package org.netbeans.modules.diff.builtin;
 import com.infradna.diff.ContextualPatch;
 import junit.framework.TestCase;
 import org.apache.commons.io.FileUtils;
-import org.apache.commons.io.IOUtils;
 
-import javax.sound.midi.Patch;
 import java.io.File;
 
 /**
@@ -15,7 +13,11 @@ public class PatchTest extends TestCase {
     public void testPatchOneFile() throws Exception {
         File p = File.createTempFile("test", "diff");
         FileUtils.copyURLToFile(getClass().getResource("singleFilePatch.diff"),p);
-        ContextualPatch p = ContextualPatch.create(p, null);
-        p.patch(false);
+
+        File b = File.createTempFile("test", "base");
+        FileUtils.copyURLToFile(getClass().getResource("base.txt"),b);
+
+        ContextualPatch patch = ContextualPatch.create(p, b);
+        patch.patch(false);
     }
 }
qgi